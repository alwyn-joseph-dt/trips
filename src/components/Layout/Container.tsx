import { Box, useMediaQuery, useTheme } from '@mui/material';
import React from 'react';

const Container = ({ children }: React.PropsWithChildren) => {
  const muiTheme = useTheme();
  const isMobileView = useMediaQuery(muiTheme.breakpoints.down('md'));

  return (
    <Box sx={{ p: isMobileView ? 0 : 4, maxWidth: 1280, m: 'auto' }}>
      {children}
    </Box>
  );
};

export default Container;
