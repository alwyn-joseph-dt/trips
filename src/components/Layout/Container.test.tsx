import React from 'react';
import { render, screen } from '@testing-library/react';
import Container from './Container';
import { useMediaQuery } from '@mui/material';

jest.mock('@mui/material', () => {
  const actualMUI = jest.requireActual('@mui/material');
  return {
    ...actualMUI,
    useMediaQuery: jest.fn(),
  };
});

describe('Container Component', () => {
  const mockUseMediaQuery = useMediaQuery as jest.Mock;

  test('renders children content', () => {
    mockUseMediaQuery.mockReturnValue(false);

    render(<Container>Test Content</Container>);

    expect(screen.getByText('Test Content')).toBeInTheDocument();
  });

  test('applies padding (desktop view)', () => {
    mockUseMediaQuery.mockReturnValue(false);

    const { container } = render(<Container>Desktop</Container>);

    expect(container.firstChild).toHaveStyle('padding: 32px');
    expect(container.firstChild).toMatchSnapshot(); // ✅ Snapshot
  });

  test('removes padding on mobile view', () => {
    mockUseMediaQuery.mockReturnValue(true);

    const { container } = render(<Container>Mobile</Container>);

    expect(container.firstChild).toHaveStyle('padding: 0px');
    expect(container.firstChild).toMatchSnapshot(); // ✅ Snapshot
  });
});
