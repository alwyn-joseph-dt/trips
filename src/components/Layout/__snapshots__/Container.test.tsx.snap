// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Container Component applies padding (desktop view) 1`] = `
<div
  class="MuiBox-root css-1ddggyt"
>
  Desktop
</div>
`;

exports[`Container Component removes padding on mobile view 1`] = `
<div
  class="MuiBox-root css-l3s00c"
>
  Mobile
</div>
`;
